src/binary-writer.cc

```diff
@@ -289,6 +298,30 @@ Index BinaryWriter::GetExceptVarDepth(const Var* var) {
   return var->index();
 }
 
+Index BinaryWriter::GetSymbolIndex(RelocType reloc_type, Index index) {
+  std::string name;
+  SymbolType type = SymbolType::Function;
+  switch (reloc_type) {
+    case RelocType::FuncIndexLEB:
+      name = module_->funcs[index]->name;
+      break;
+    case RelocType::GlobalIndexLEB:
+      type = SymbolType::Global;
+      name = module_->globals[index]->name;
+      break;
+    default:
+      WABT_UNREACHABLE;
+  }
+  auto iter = symtab_.find(name);
+  if (iter != symtab_.end())
+    return iter->second;
+
+  Index sym_index = Index(symbols_.size());
+  symtab_[name] = sym_index;
+  symbols_.push_back(Symbol{sym_index, type, index});
```

Comment:
`emplace_back` instead?

Reply:
I don't think that works with POD structs